src/main/java/org/motechproject/mots/service/UserLogService.java

```diff
@@ -26,4 +27,22 @@ public UserLog createNewUserLog(User user, Date expirationDate) {
     userLog.setLogoutDate(expirationDate);
     return userLogRepository.save(userLog);
   }
+
+  public UserLog updateUserLog(UserLog userLog) {
+    return userLogRepository.save(userLog);
+  }
+
+  /**
+   * @param username of current user
+   * @return most recent user's user log.
+   */
+  public UserLog getUserLog(String username) {
+    List<UserLog> userLogs = userLogRepository
+        .findByUser_UsernameAndLogoutDateGreaterThanEqualOrderByLogoutDateDesc(username,
+            new Date());
```

Comment:
Let's use (current date - tokenValiditySeconds). We will usually request refresh tokens after the original access token has expired, so the current date will be past logoutDate.
You can get the token validity seconds like so:
`@Value("${token.validitySeconds}")
  private Integer tokenValiditySeconds;`

Reply:
Come to think of it - it would be better to order UserLogs by login date and remove the date check. Just make sure that we are only using this method when refreshing tokens.

Reply:
findFirstByUsernameOrderByLoginDateDesc()